odl/util/ufuncs.py

```diff
@@ -335,10 +334,15 @@ def wrapper(self, out1=None, out2=None):
     elif n_in == 2:
         if n_out == 1:
             def wrapper(self, x2, out=None):
-                try:
+                if x2 in self.vector.space:
                     x2 = x2.ntuple
+
+                try:
+                    # Try to reshape to linear data
+                    x2 = x2.reshape(self.vector.size,
+                                    order=self.vector.space.order)
                 except AttributeError:
```

Comment:
Well `NtuplesBaseVector` is the base case, they all go down here, this is admitedly a hack to allow users to for example add ndarrays to DiscreteLp guys, perhaps we can move this to that part.